File: Python/242_Valid_Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        letterdict={}
        for c in s:
            if c in letterdict.keys():
                letterdict[c]+=1
            else:
                letterdict[c]=1
        for c in t:
            if c in letterdict.keys() and letterdict[c]>0:
                letterdict[c]-=1
            else:
                return False
        for i in letterdict.keys():
            if letterdict[i]>0:
                return False
        return True

File: Java/242_Valid_Anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()){
            return false;
        } 
        int[] temp=new int[26];
        for (int i=0;i<s.length();i++){
            temp[s.charAt(i)-'a']++;
        }
        for (int j=0;j<t.length();j++){
            if (temp[t.charAt(j)-'a']<=0){
                return false;
            }
            temp[t.charAt(j)-'a']--;
        }
        return true;       

    }
}  